.github/workflows/build-test-osx-m1-cirrus.yaml

```diff
@@ -0,0 +1,142 @@
+# This workflow builds and tests the semgrep-core binary for macOS M1
```

Comment:
Please add a toplevel comment explaining what this workflow is about. I have no idea what
Cirrus is and why we need it instead of what we are currently using.